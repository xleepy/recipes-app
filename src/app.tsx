import { Route, Routes } from 'react-router-dom';
import { Home } from './pages/home';
import { AppProviders } from './providers/app-providers';
import { lazy } from 'react';

const Detail = lazy(() => import('./pages/detail'));

export function App() {
  return (
    <AppProviders>
      <Routes>
        <Route element={<Home />} index />
        <Route path="/detail/:id" element={<Detail />} />
      </Routes>
    </AppProviders>
  );
}
