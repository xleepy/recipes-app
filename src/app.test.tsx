import { render } from '@testing-library/preact';
import { describe, expect, it } from 'vitest';
import { App } from './app';

// skipped till issue will be resolved
// https://github.com/vitest-dev/vitest/issues/5004
describe.skip('App tests', () => {
  it('should render app', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
