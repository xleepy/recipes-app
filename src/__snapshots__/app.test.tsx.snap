// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App tests > should render app 1`] = `
<DocumentFragment>
  <div>
    <input
      class="_search_4d1a47"
      data-testid="search"
      placeholder="Enter query"
      type="text"
      value="pasta"
    />
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
